src/IsoCodes/ZipCode.php

```diff
@@ -21,111 +217,84 @@ public static function validate($zipcode, $country)
         if (empty($zipcode)) {
             return false;
         }
+
+        $country = strtoupper($country);
+        if (isset(self::$patterns[$country])) {
+            $regexp = '/^'.self::$patterns[$country].'$/';
+
+            return (boolean) preg_match($regexp, $zipcode);
+        }
+
         $methodName = 'validate'.trim(ucfirst(strtolower($country)));
         if (!is_callable(array(__CLASS__, $methodName))) {
             throw new \InvalidArgumentException("ERROR: The zipcode validator for $country does not exists yet: feel free to add it.");
         }
-
         return call_user_func_array(array(__CLASS__, $methodName), array($zipcode));
     }
 
     /**
-     * US "ZIP+4" zipcode validator.
-     *
-     * @param string $zipcode
+     * @param $zipcode
      *
      * @return bool
+     * @deprecated use validate($zipcode, 'US')
```

Comment:
Nice Improvement, thanks for that.

OK for @Soullivaneuh proposal: could you please add this `E_USER_DEPRECATED` error trigger to the @deprecated blocks (as [in use in Symfony2 codebase](https://github.com/symfony/symfony/search?utf8=%E2%9C%93&q=trigger_error) by example)

As proposed, `1.2` starts the deprecation, and `2.0` will remove this.

```php
trigger_error('The '.__METHOD__.' method is deprecated since version 1.2 and will be removed in 2.0. You should use the validate($zipcode, $country) method instead.', E_USER_DEPRECATED);
```